luarules/gadgets/unit_targetting_priorities.lua

```diff
@@ -16,12 +16,33 @@ if gadgetHandler:IsSyncedCode() then
     local EditUnitCmdDesc = Spring.EditUnitCmdDesc
     local FindUnitCmdDesc = Spring.FindUnitCmdDesc
     local GetUnitDefID = Spring.GetUnitDefID
-    local GetUnitRulesParam = Spring.GetUnitRulesParam
     local InsertUnitCmdDesc = Spring.InsertUnitCmdDesc
     local SetUnitRulesParam = Spring.SetUnitRulesParam
+    local GetUnitRulesParam = Spring.GetUnitRulesParam
+    local GetUnitTeam = Spring.GetUnitTeam
     local CMD_SET_PRIORITY = 34567
+    local PRIORITY_GOOD = 0.001
+    local PRIORITY_BAD = 100
+    local PRIORITY_INDIFFERENT = 1
+    local PRIORITY_MUST_NOT = -1
+
+    local NO_PRIORITY_INDEX = 0
+
+    local unitToPriorityMap = {}
+    local playerPriorityCount = {}
+    local PLAYER_PRIORITY_LIMIT
+    -- maybe the modoptions should be available as a global table
+    -- which also uses the modoption keys as keys
+    for _, v in pairs(VFS.Include("modoptions.lua")) do
+        if v.key == "mo_priority_unit_limit" then
+            PLAYER_PRIORITY_LIMIT = v.def
+            break
+        end
+    end

```

Comment:
This is just to retrieve the default value set in modoptions, instead of hard-coding it twice. Other gadgets could make use of having modoptions in a global table as well.